environment.py

```diff
@@ -64,14 +64,23 @@ def env() -> Mapping[str, Optional[str]]:
         # TDR source = 'tdr:', Google Cloud project name,
         #              ':', ( 'dataset' | 'snapshot' ),
         #              '/', TDR dataset or snapshot name,
-        #              ':', [ UUID prefix [ '/', Partition prefix length ] ] ;
+        #              ':', [ UUID prefix ], [ '/', partition prefix length ] ;
         #
         # canned source = 'https://github.com',
         #                 '/', owner,
         #                 '/', repo,
         #                 '/tree/', ref,
-        #                 ['/', path] ;
-        #                 ':', [ UUID prefix [ '/', Partition prefix length ] ] ;
+        #                 ['/', path],
+        #                 ':', [ UUID prefix ], [ '/', partition prefix length ] ;
+        #
+        # The UUID prefix is an optional string of hexadecimal digits that
+        # constrains the subgraphs that are indexed within a source.
+        # The partition prefix length is an optional integer that controls how
+        # many reindex notifications are created for the indexer lambdas to
+        # process. The number of notifications created is
+        # 16 ** `partition prefix length`. Due to fixed bits in a version 4
+        # UUID, the sum of the length for a UUID prefix and the partition
+        # prefix length must be less than 8.
```

Comment:
The bits are fixed in all versions. The length can be 8. The paragraph has both too much detail and some detail missing.

```suggestion
        # The `prefix` is an optional string of hexadecimal digits
        # constraining the set of indexed subgraphs from the source. A
        # subgraph will be indexed if its UUID begins with the `prefix`. The
        # default `prefix` is the empty string.
        # 
        # The partition prefix length is an optional integer that is used to
        # further partition the set of indexed subgraphs. Each partition is
        # assigned a prefix of `partition prefix length` hexadecimal digits.
        # A subgraph belongs to a partition if its UUID starts with the
        # overall `prefix` followed by the partition's prefix. The number of
        # partitions of a source is therefore `16 ** partition prefix
        # length`. The default `partition prefix length` is configured
        # in ...
        # 
        # The `partition prefix length` plus the length of `prefix` must not
        # exceed 8.  
```